main/apps.js

```diff
@@ -113,6 +114,34 @@ function generateUpdatesJsonFile() {
 }
 
 /**
+ * Show confirmation dialog about existing app directory to be removed.
+ *
+ * @param {string} tgzFilePath Path to the tgz file to install.
+ * @param {string} appPath Path of the existing app directory.
+ * @returns {Promise} promise that resolves when existing app has been removed.
+ */
+function confirmAndRemoveOldLocalApp(tgzFilePath, appPath) {
+    return new Promise((resolve, reject) => {
+        dialog.showMessageBox({
+            type: 'question',
+            title: 'Existing app directory',
+            message: `Tried to extract archive ${tgzFilePath}, ` +
+                `but app directory ${appPath} already exists.\n\n` +
+                'Do you want to remove existing app in order to extract the archive?',
+            buttons: ['Remove', 'Cancel'],
+        }, btnIndex => {
+            if (btnIndex === 1) {
+                return reject(new Error(`Tried to extract archive ${tgzFilePath}, ` +
+                    `but app directory ${appPath} already exists. Either delete the ` +
+                    'app directory so that the archive can be extracted, or delete ' +
+                    'the archive file.'));
```

Comment:
How about proceeding without removing/extracting if the user cancels?

Reply:
Good point, I thought maybe a choice to remove either of the offending articles, and also in case of cancel just proceed, but then next time same situation...